### PR TITLE
Implementation multiple blocking

### LINKED ISSUE #313: Multiple blocks
- [x] List of who rated the account
- [x] In the block transaction, make it possible to accept an array

### PR DESCRIPTION
**Pull request by only1question**
Implementation multiple blocking

### CODE REVIEW

File: src/pocketdb/consensus/social/Blocking.hpp [hunk 1]
```diff
@@ -86,14 +86,101 @@ namespace PocketConsensus
...
+                    continue;
+
+                auto blockPtx = static_pointer_cast<Blocking>(blockTx);
+
+                if (*blockPtx->GetHash() == *ptx->GetHash())
+                    continue;
+
+                if (*ptx->GetAddress() == *blockPtx->GetAddress() && *ptx->GetAddressTo() == *blockPtx->GetAddressTo() &&
+                    *ptx->GetAddressesTo() == *blockPtx->GetAddressesTo())
+                    return {false, SocialConsensusResult_ManyTransactions};
```
Comment (andyoknen) [inline src/pocketdb/consensus/social/Blocking.hpp]: need check IsEmpty(GetAddressTo) and IsEmpty(GetAddressesTo) for ptx and blockPtx

File: src/pocketdb/models/dto/Blocking.cpp [hunk 2]
```diff
@@ -55,6 +60,16 @@ namespace PocketTx
...
         string data;
         data += GetAddressTo() ? *GetAddressTo() : "";
+        if (GetAddressesTo())
+        {
+            UniValue addrs(UniValue::VARR);
+            addrs.read(*GetAddressesTo());
+            for (size_t i = 0; i < addrs.size(); ++i)
+            {
+                data += (i ? "," : "");
+                data += addrs[i].get_str();
```
Comment (andyoknen) [inline src/pocketdb/models/dto/Blocking.cpp:70]: `data += (i ? "," : "");` - no need?

File: src/pocketdb/consensus/social/Blocking.hpp (hunk 1 above)
Comment (andyoknen) [reply src/pocketdb/consensus/social/Blocking.hpp]: Also situation:
- In mempool exist transaction A -> B
- New transaction A -> [ B ]

or on the contrary

File: src/pocketdb/consensus/social/BlockingCancel.hpp [hunk 3]
```diff
@@ -86,6 +86,30 @@ namespace PocketConsensus
...
+        {
+            // Base validation with calling block or mempool check
+            if (auto[baseValidate, baseValidateCode] = SocialConsensus::Validate(tx, ptx, block); !baseValidate)
+                return {false, baseValidateCode};
+
+            if (!PocketDb::ConsensusRepoInst.ExistBlocking(
+                *ptx->GetAddress(),
+                IsEmpty(ptx->GetAddressTo()) ? "" : *ptx->GetAddressTo(),
+                "[]"
+            ))
```
Comment (andyoknen) [inline src/pocketdb/consensus/social/BlockingCancel.hpp:102]: if in mempool exists tx 305 A > [ B ]
and new tx 306 A > B
?

File: src/pocketdb/consensus/social/Blocking.hpp (hunk 1 above)
Comment (only1question) [reply src/pocketdb/consensus/social/Blocking.hpp]: Edited

File: src/pocketdb/models/dto/Blocking.cpp (hunk 2 above)
Comment (only1question) [reply src/pocketdb/models/dto/Blocking.cpp:70]: It does not matter

File: src/pocketdb/migrations/main.cpp [hunk 4]
```diff
@@ -182,12 +182,41 @@ namespace PocketDb
...
+              ut.Id
+            from Transactions b indexed by Transactions_Type_Last_Height_Id
+            join Transactions us indexed by Transactions_Type_Last_String1_Height_Id
+              on us.Type in (100) and us.Last = 1 and us.String1 = b.String1 and us.Height > 0
+            join Transactions ut indexed by Transactions_Type_Last_String1_Height_Id
+              on ut.Type in (100) and ut.Last = 1 and ut.String1 = b.String2 and ut.Height > 0
+            where b.Type in (305)
+              and b.Last = 1
+              and b.Height > 0
+              and not exists (select 1 from BlockingLists);
```
Comment (andyoknen) [inline src/pocketdb/migrations/main.cpp]: -----

**Review by andyoknen — APPROVED** [review]

**Review by lostystyg — APPROVED** [review]

**Pull request merged**